aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -182,6 +182,13 @@ module aptos_framework::transaction_context {
         payload.entry_function_payload
     }
 
+    /// Returns the hash of the current raw transaction, which is used for transaction authentication.
+    /// This function calls an internal native function to retrieve the raw transaction hash.
+    public fun raw_txn_hash(): vector<u8> {
```

Comment:
what will this be for non-user transactions? will it abort or returning something?
is there any uniqueness guarantees?

(irrespectively, function should be raw_txn_hash() -> get_raw_txn_hash() to follow the naming of other functions)

Reply:
> what will this be for non-user transactions? will it abort or returning something?
It will abort when it's a non-user transaction and a user transaction context is unavailable. https://github.com/aptos-labs/aptos-core/blob/71ec744ea22fa5566ecb37f65485fce268e85e70/aptos-move/framework/aptos-framework/sources/transaction_context.move#L273

> is there any uniqueness guarantees?
The returned hash values should differ if the raw transactions are different. However, it will remain the same across sessions (prologue, execution, and epilogue). In that sense, it's different from the `get_transaction_hash` function.

> (irrespectively, function should be raw_txn_hash() -> get_raw_txn_hash() to follow the naming of other functions)
The convention seems to avoid using the `get_` prefix, except for `get_transaction_hash` and `get_script_hash`.
https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/transaction_context.move#L71-L183

Reply:
what's the hash from `get_transaction_hash`?

Reply:
It's the hash of the `SessionId`, which is an internal data structure within the Aptos VM.

Reply:
could we add a feature flag and reuse this function?

Reply:
get_transaction_hash is used extensively with the assumption it is a globally unique identifier to the session (across all transaction types)

new one isn't 

so it will break anything that depends on it.

Naming here is very poor. Maybe we should deprecate this one (with `#[deprecated]`), and add two new functions with clearer names, then compiler will warn people - and we can update in framework accordingly.

we can name them `transaction_hash` and `unique_session_hash`

Reply:
> we can name them transaction_hash and unique_session_hash

How about `raw_transaction_hash`? It's a bit more descriptive and less likely to cause confusion with other types of transaction hashes in the future.